tests/lib/reset.sh

```diff
@@ -138,25 +145,37 @@ reset_all_snap() {
     fi
 }
 
-if is_core_system; then
-    reset_all_snap "$@"
-else
-    reset_classic "$@"
-fi
-
-# Discard all mount namespaces and active mount profiles.
-# This is duplicating logic in snap-discard-ns but it doesn't
-# support --all switch yet so we cannot use it.
-if [ -d /run/snapd/ns ]; then
-    for mnt in /run/snapd/ns/*.mnt; do
-        umount -l "$mnt" || true
-        rm -f "$mnt"
-    done
-    rm -f /run/snapd/ns/*.fstab
-fi
-
-if [ "$REMOTE_STORE" = staging ] && [ "$1" = "--store" ]; then
-    # shellcheck source=tests/lib/store.sh
-    . "$TESTSLIB"/store.sh
-    teardown_staging_store
-fi
+discard_ns() {
+    # Discard all mount namespaces and active mount profiles.
+    # This is duplicating logic in snap-discard-ns but it doesn't
+    # support --all switch yet so we cannot use it.
+    if [ -d /run/snapd/ns ]; then
+        for mnt in /run/snapd/ns/*.mnt; do
+            umount -l "$mnt" || true
+            rm -f "$mnt"
+        done
+        rm -f /run/snapd/ns/*.fstab
+    fi
+}
+
+tear_down_store() {
+    if [ "$REMOTE_STORE" = staging ] && [ "$1" = "--store" ]; then
+        # shellcheck source=tests/lib/store.sh
+        . "$TESTSLIB"/store.sh
+        teardown_staging_store
+    fi  
+}
+
+reset_snapd() {
+    if is_core_system; then
```

Comment:
Since this is a function you can move `discard_ns` and `tear_down_store "$@" to after the if-then-else statement.

Reply:
done